File: base/src/main/java/com/ui/base/FragmentParentActivity.java
package com.ui.base;

import android.app.Activity;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.ui.R;

public class FragmentParentActivity extends BaseActivity {

    public static void startActivity(Activity context, Class<?> clz, boolean isToolbar) {
        Intent intent = new Intent(context, FragmentParentActivity.class);
        intent.putExtra(FragmentParentActivity.KEY_FRAGMENT, clz);
        intent.putExtra(FragmentParentActivity.KEY_TOOLBAR, isToolbar);
        context.startActivity(intent);
    }

    public static void startActivity(Activity context, Class<?> clz) {
        Intent intent = new Intent(context, FragmentParentActivity.class);
        intent.putExtra(FragmentParentActivity.KEY_FRAGMENT, clz);
        context.startActivity(intent);
    }

    public final static String KEY_FRAGMENT_NAME = "KEY_FRAGMENT_NAME";

    public final static String KEY_FRAGMENT = "KEY_FRAGMENT";
    public final static String KEY_TOOLBAR = "KEY_TOOLBAR";
    private BaseFragment baseFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        boolean isToolbar = getIntent().getBooleanExtra(KEY_TOOLBAR, true);
        Class<?> clz = (Class<?>) getIntent().getSerializableExtra(KEY_FRAGMENT);
        String cls;
        if (clz != null) {
            cls = clz.getName();
        } else {
            cls = getIntent().getStringExtra(KEY_FRAGMENT_NAME);
        }

        if (isToolbar) {
            setContentView(R.layout.activity_with_toolbar_layout);
            Fragment fragment = Fragment.instantiate(getActivity(), cls);
            if (fragment instanceof BaseFragment)
                baseFragment = (BaseFragment) fragment;
            FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
            ft.replace(R.id.frame_holder, fragment, cls);
            ft.commitAllowingStateLoss();
        } else {
            setRootView((ViewGroup) getWindow().getDecorView());
            initProgressLayout();
            Fragment fragment = Fragment.instantiate(getActivity(), cls);
            if (fragment instanceof BaseFragment)
                baseFragment = (BaseFragment) fragment;
            FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
            ft.replace(android.R.id.content, fragment, cls);
            ft.commitAllowingStateLoss();
        }
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (baseFragment != null)
            baseFragment.onActivityResult(requestCode, resultCode, data);
    }

    @Override
    public void onConfigurationChanged(@NonNull Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }

}

File: app/src/main/java/com/song/sakura/ui/base/FragmentParentActivity.kt
package com.song.sakura.ui.base

import android.app.Activity
import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.text.TextUtils
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.ui.base.BaseActivity
import com.gyf.immersionbar.ImmersionBar
import com.song.sakura.R

class FragmentParentActivity : BaseActivity() {


    fun initImmersionBar() {
        ImmersionBar.with(this)!!.statusBarDarkFont(true).init()
    }

    fun initImmersionBar2() {
        ImmersionBar.with(this)!!.titleBar(R.id.appbar).statusBarColorInt(getColors(R.color.white))
                .statusBarDarkFont(true).init()
    }

    public override fun onDestroy() {
        super.onDestroy()
    }

    private var baseFragment: IBaseFragment<*>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val isImmbar = intent.getBooleanExtra(KEY_IMMBAR, true)
        val isToolbar = intent.getBooleanExtra(KEY_TOOLBAR, true)
        val clz = intent.getSerializableExtra(KEY_FRAGMENT) as Class<*>
        var cls = ""
        if (TextUtils.isEmpty(cls)) {
            cls = clz.name
        } else {
            cls = intent.getStringExtra(KEY_FRAGMENT_NAME)
        }


        if (isToolbar) {
            setContentView(R.layout.activity_with_toolbar_layout)
            if (isImmbar) {
                initImmersionBar2()
            }
            val fragment = Fragment.instantiate(activity, cls)
            if (fragment is IBaseFragment<*>)
                baseFragment = fragment
            val ft = supportFragmentManager.beginTransaction()
            ft.replace(R.id.frame_holder, fragment, cls)
            ft.commitAllowingStateLoss()
        } else {
            setRootView(window.decorView as ViewGroup)
            initProgressLayout()
            val fragment = Fragment.instantiate(activity, cls)
            if (fragment is IBaseFragment<*>)
                baseFragment = fragment
            val ft = supportFragmentManager.beginTransaction()
            ft.replace(android.R.id.content, fragment, cls)
            ft.commitAllowingStateLoss()
            if (isImmbar) {
                initImmersionBar()
            }
        }

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (baseFragment != null)
            baseFragment!!.onActivityResult(requestCode, resultCode, data)
    }


    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
    }

    companion object {


        fun startActivity(context: Activity, clz: Class<*>, isToolbar: Boolean) {
            val intent = Intent(context, FragmentParentActivity::class.java)
            intent.putExtra(KEY_FRAGMENT, clz)
            intent.putExtra(KEY_TOOLBAR, isToolbar)
            context.startActivity(intent)
        }


        fun startActivity(context: Activity, clz: Class<*>) {
            val intent = Intent(context, FragmentParentActivity::class.java)
            intent.putExtra(KEY_FRAGMENT, clz)
            context.startActivity(intent)
        }

        val KEY_FRAGMENT_NAME = "KEY_FRAGMENT_NAME"

        const val KEY_FRAGMENT = "KEY_FRAGMENT"
        const val KEY_TOOLBAR = "KEY_TOOLBAR"
        const val KEY_IMMBAR = "KEY_IMMBAR"
    }

}
